filesystem/iso9660.ksy

```diff
@@ -7,269 +7,959 @@ meta:
+                doc-ref: ecma-119 9.1.6 b7
+                type: b1
+              - id: file_flags_reserved
+                doc-ref: ecma-119 9.1.6 b5+b6
+                type: b2
+              - id: file_flags_protection
+                doc-ref: ecma-119 9.1.6 b4
+                type: b1
+              - id: file_flags_record
+                doc-ref: ecma-119 9.1.6 b3
+                type: b1
+              - id: file_flags_associated_file
+                doc-ref: ecma-119 9.1.6 b2
+                type: b1
+              - id: file_flags_directory
+                doc-ref: ecma-119 9.1.6 b1
+                type: b1
+              - id: file_flags_existence
+                doc-ref: ecma-119 9.1.6 b0
+                type: b1
+              - id: file_unit_size
+                doc-ref: ecma-119 9.1.7
+                type: u1
+              - id: interleave_gap_size
+                doc-ref: ecma-119 9.1.8
+                type: u1
+              - id: vol_seq_num
+                doc-ref: ecma-119 9.1.9
+                type: u2bi
+              - id: len_fi
+                doc-ref: ecma-119 9.1.10
+                type: u1
+              - id: file_id_file
+                doc-ref: ecma-119 9.1.11
+                type: str
+                size: len_fi
+                if: not file_flags_directory
+              - id: file_id_dir
+                doc-ref: ecma-119 9.1.11
+                size: len_fi
+                type: str
+                if: file_flags_directory
+              - id: padding_field
+                doc-ref: ecma-119 9.1.12
+                doc: |
+                  Padding field is added when len_fi contains an even number
+                contents: [ 0x0 ]
+                if: len_fi % 2 == 0
+              - id: system_use
+                doc-ref: ecma-119 9.1.13
+                doc: |
+                  The size of the system_use is defined as:
+                  ( len_dr - size of directory_record = 33 bytes ) - len_fi
+                type: susp
+                size: ( _parent.len_dr - 33 ) - len_fi
+            instances:
+              directory_records:
+                io: _root._io
+                pos: _root.sector_size * location_of_extent.le
+                size: data_len.le
+                type: directory_records
+                if: ( _parent.len_dr > 0x0 ) and file_flags_directory
+              file_content:
+                io: _root._io
+                pos: _root.sector_size * location_of_extent.le
+                size: data_len.le
+                if: ( _parent.len_dr > 0x0 ) and not file_flags_directory
+            types:
+              susp:
+                doc: |
+                  We check if we have at least 2 bytes left.
+                  The SUSP magic is 2 bytes.
+                seq:
+                  - id: header
+                    type: header
+                    if: _io.size - _io.pos >= 2
+                    repeat: eos
```

Comment:
According to IEEE P1281 a valid SUSP entry is at least 4 bytes long:

```
If the remaining allocated space following the last recorded System Use Entry in a System Use field or Continuation Area is less than four bytes long, it cannot contain a System Use Entry and shall be ignored.
```
I have adapted `susp` to the following:

```
              susp:
                doc: |
                  We check if we have at least 4 bytes left.
                  The SUSP magic is 2 bytes. A complete SUSP entry
                  is at least 4 bytes long.
                seq:
                  - id: header
                    type: header
                    if: _io.size - _io.pos >= 4
                    repeat: until
                    repeat-until: _io.size - _io.pos < 4

```